README.md
# NAME

DBIx::Class::Sweeten - Short intro

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.10.1+-blue.svg" alt="Requires Perl 5.10.1+" />
    <a href="https://travis-ci.org//"><img src="https://api.travis-ci.org//.svg?branch=master" alt="Travis status" /></a>
    <a href="http://cpants.cpanauthors.org/release/CSSON/DBIx-Class-Sweeten-0.0103"><img src="http://badgedepot.code301.com/badge/kwalitee/CSSON/DBIx-Class-Sweeten/0.0103" alt="Distribution kwalitee" /></a>
    <a href="http://matrix.cpantesters.org/?dist=DBIx-Class-Sweeten%200.0103"><img src="http://badgedepot.code301.com/badge/cpantesters/DBIx-Class-Sweeten/0.0103" alt="CPAN Testers result" /></a>
    </p>
</div>

# VERSION

Version 0.0103, released 2018-02-10.

# SYNOPSIS

    # in MyApp::Schema, instead of inheriting from DBIx::Class::Schema
    use base 'DBIx::Class::Sweeten::Schema';

# DESCRIPTION

DBIx::Class::Sweeten is a collection of modules that can be used to reduce some boilerplate associated with [DBIx::Class](https://metacpan.org/pod/DBIx::Class).

- [DBIx::Class::Sweeten::Schema](https://metacpan.org/pod/DBIx::Class::Sweeten::Schema) - Access resultsets via method calls
- [DBIx::Class::Sweeten::Result::Base](https://metacpan.org/pod/DBIx::Class::Sweeten::Result::Base) - Adds a column attribute for easier indexing
- [DBIx::Class::Sweeten::Result](https://metacpan.org/pod/DBIx::Class::Sweeten::Result) - DBIx::Class::Candy defaults for result sources
- [DBIx::Class::Sweeten::ResultSet](https://metacpan.org/pod/DBIx::Class::Sweeten::ResultSet) - DBIx::Class::Candy defaults for resultsets

# SEE ALSO

- [DBIx::Class](https://metacpan.org/pod/DBIx::Class)
- [DBIx::Class::Candy](https://metacpan.org/pod/DBIx::Class::Candy) - Sugar for `DBIx::Class`
- [DBIx::Class::Helpers](https://metacpan.org/pod/DBIx::Class::Helpers) - More `DBIx::Class` goodness
- [DBICx::Shortcuts](https://metacpan.org/pod/DBICx::Shortcuts) - An alternative for `DBIx::Class::Sweeten::Schema`

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for DBICx-Sweeten

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.0103 2018-02-10T10:37:56Z UTC
 [Bug Fixes]
 - ::Schema overrides 'connection' instead of 'connect'. It now passes the
   upcoming DBIx::Class sanity checker.

 [New Features]
 - ::Tidy, with a better API
 - ::ResultSet::Base, with filter() and filterattr(), instead of search({},
   {})

 [Stats & Dependency Changes]
 - 12 files changed, 911 insertions(+), 17 deletions(-)
 - (run req) ~ perl 5.016000 --> 5.010001
 - (dev req) + Dist::Zilla::Plugin::MetaConfig (any)
 - (dev req) + Dist::Zilla::Plugin::PromptIfStale 0.054
 - (dev req) ~ Dist::Zilla 5.000 --> 6.010
 - (dev req) ~ Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007 --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0100
   --> 0.0200
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild 0.151680 -->
   0.170880
 - (dev req) ~ Dist::Zilla::Plugin::Git 2.036 --> 2.041
 - (dev req) ~ Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013 -->
   0.014
 - (dev req) ~ Dist::Zilla::Plugin::Git::Contributors 0.017 --> 0.030
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package 2.003001 -->
   2.004003
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod 0.150250 --> 0.163250
 - (dev req) ~ Dist::Zilla::Plugin::Test::Compile 2.054 --> 2.057
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs 0.021 --> 0.027
 - (dev req) - Dist::Zilla::Plugin::Test::Kwalitee::Extra
 - (dev sug) ~ Dist::Iller::Config::Author::CSSON 0.0309 --> 0.0319

0.0101 2016-03-19T20:43:55Z UTC
 [Stats & Dependency Changes]
 - 7 files changed, 985 insertions(+), 81 deletions(-)
 - (run req) + DBIx::Class (any)
 - (run req) + DBIx::Class::Candy (any)
 - (run req) + String::CamelCase 0.02
 - (test req) + IO::Handle (any)
 - (test req) + IPC::Open3 (any)
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Badge::Depot::Plugin::Travis (any)
 - (dev req) + Dist::Zilla 5.000
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.007
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0100
 - (dev req) + Dist::Zilla::Plugin::Clean 0.07
 - (dev req) + Dist::Zilla::Plugin::DistIller::MetaGeneratedBy (any)
 - (dev req) + Dist::Zilla::Plugin::Git 2.036
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.013
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Contributors 0.017
 - (dev req) + Dist::Zilla::Plugin::InstallRelease 0.008
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::Test::Compile 2.054
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::Kwalitee::Extra v0.2.1
 - (dev req) + Dist::Zilla::Plugin::Test::NoTabs 0.15
 - (dev req) + Dist::Zilla::Plugin::Test::Version 1.05
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::EOL (any)
 - (dev req) + Test::More 0.88
 - (dev req) + Test::NoTabs 1.4
 - (dev req) + Test::Version 1
 - (dev req) ~ Dist::Zilla::Plugin::CheckChangesHasContent (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild (any) --> 0.151680
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package (any) -->
   2.003001
 - (dev req) ~ Dist::Zilla::Plugin::PodWeaver (any) --> 4.006
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod (any) --> 0.150250
 - (dev req) ~ Dist::Zilla::Plugin::RunExtraTests (any) --> 0.028
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs (any) --> 0.021
 - (dev req) ~ Dist::Zilla::Plugin::TravisYML (any) --> 1.15
 - (dev req) ~ Dist::Zilla::Plugin::UploadToStratopan (any) --> 0.012
 - (dev req) ~ Pod::Weaver::Section::Badges (any) --> 0.0401
 - (dev req) ~ Test::Warnings (any) --> 0.026
 - (dev req) - Dist::Zilla::Plugin::CheckExtraTests
 - (dev req) - Dist::Zilla::Plugin::ContributorsFromGit
 - (dev req) - Dist::Zilla::Plugin::MetaProvides::Class
 - (dev req) - Dist::Zilla::Plugin::NextRelease
 - (dev req) - Dist::Zilla::Plugin::NextVersion::Semantic
 - (dev req) - Dist::Zilla::Plugin::OurPkgVersion
 - (dev req) - Dist::Zilla::Plugin::PodnameFromClassname
 - (dev req) - Dist::Zilla::Plugin::Prereqs::Plugins
 - (dev req) - Dist::Zilla::Plugin::PreviousVersion::Changelog
 - (dev req) - Dist::Zilla::Plugin::ReversionOnRelease
 - (dev req) - Pod::Weaver::Section::Homepage::DefaultCPAN
 - (dev req) - Pod::Weaver::Section::Source::DefaultGitHub
 - (dev sug) + Dist::Iller 0.1408
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0309

0.0100 2015-11-07T17:14:32Z UTC
 - Initial release


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.010.
use strict;
use warnings;

use 5.010001;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Short intro",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "DBIx-Class-Sweeten",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010001",
  "NAME" => "DBIx::Class::Sweeten",
  "PREREQ_PM" => {
    "DBIx::Class" => 0,
    "DBIx::Class::Candy" => 0,
    "String::CamelCase" => "0.02"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.0104",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "DBIx::Class" => 0,
  "DBIx::Class::Candy" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "String::CamelCase" => "0.02",
  "Test::More" => "0.96"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
